### ISSUE TITLE
[Proposal] Equal height for all pinned repositories

### ISSUE THREAD

**Issue by razonyang** [anuraghazra/github-readme-stats#2900]
### Is your feature request related to a problem? Please describe.

Hi, I found that the pinned repos are not equal height.
![image](https://github.com/anuraghazra/github-readme-stats/assets/17720932/d59a59bd-b5be-420b-b2e6-4054cbf46020)

### Describe the solution you'd like

GitHub built-in:
![image](https://github.com/anuraghazra/github-readme-stats/assets/17720932/575b329e-591c-4a7f-b41d-cd360e39fab2)

**Comment by rickstaa** [comment]
@razonyang, thanks for your bug report. This is the intended behaviour since we want the card to resize to the repository description automatically, and users are expected to handle the card height themselves. The GitHub pin card this card was based on handling the height change on the profile page.

![image](https://github.com/anuraghazra/github-readme-stats/assets/17570430/bc162c66-db23-45f3-8d26-75dc63ab3e2c)

So I think I prefer to close this issue. However, if @qwerty541 and @anuraghazra think this behaviour is not intended, we can change the minimum size of the `repo/pin` card to show at least two lines. 

```html
<a href="https://github.com/anuraghazra/github-readme-stats">
  <img align="center" src="https://github-readme-stats.vercel.app/api/pin/?username=anuraghazra&repo=github-readme-stats" />
</a>
<a href="https://github.com/anuraghazra/convoychat">
  <img align="center" src="https://github-readme-stats.vercel.app/api/pin/?username=anuraghazra&repo=convoychat" />
</a>
```

<a href="https://github.com/anuraghazra/github-readme-stats">
  <img align="center" src="https://github-readme-stats.vercel.app/api/pin/?username=anuraghazra&repo=github-readme-stats" />
</a>
<a href="https://github.com/anuraghazra/convoychat">
  <img align="center" src="https://github-readme-stats.vercel.app/api/pin/?username=anuraghazra&repo=convoychat" />
</a>

**Comment by razonyang** [comment]
@rickstaa Thanks for the code snippet, but the fixed height doesn't look good, equal width and height looks better. The font-size are also smaller than others.

![image](https://github.com/anuraghazra/github-readme-stats/assets/17720932/859da9b2-112f-4dad-9fb0-8b936dd5b60d)

It may better that if the description can take 2 rows, fill up with whitespace if not enough. Please feel free to close it if it's not possible or reasonable. I've solved it by modifying the descriptions.

**Comment by rickstaa** [comment]
> @rickstaa Thanks for the code snippet, but the fixed height doesn't look good, equal width and height looks better. The font-size are also smaller than others.
> 
> ![image](https://user-images.githubusercontent.com/17720932/250493185-859da9b2-112f-4dad-9fb0-8b936dd5b60d.png)
> 
> It may better that if the description can take 2 rows, fill up with whitespace if not enough. Please feel free to close it if it's not possible or reasonable. I've solved it by modifying the descriptions.

Thanks for your feedback. We have the [card_width](https://github.com/anuraghazra/github-readme-stats#stats-card-exclusive-options) option to change the card width. One way to improve this behaviour is to add a `card_height` option. Freel free to create a pull request 👍🏻.

**Comment by rickstaa** [comment]
> > @rickstaa Thanks for the code snippet, but the fixed height doesn't look good, equal width and height looks better. The font-size are also smaller than others.
> > ![image](https://user-images.githubusercontent.com/17720932/250493185-859da9b2-112f-4dad-9fb0-8b936dd5b60d.png)
> > It may better that if the description can take 2 rows, fill up with whitespace if not enough. Please feel free to close it if it's not possible or reasonable. I've solved it by modifying the descriptions.
> 
> Thanks for your feedback. We have the [card_width](https://github.com/anuraghazra/github-readme-stats#stats-card-exclusive-options) option to change the card width. One way to improve this behaviour is to add a `card_height` option. Freel free to create a pull request 👍🏻.

It looks like my previous comment was incorrect 🤔. The `card_width` argument has not yet been implemented for the `pin-card`. Feel free to create a pull request or open a new pull request to have this argument and the `card_height` argument implemented.

**Comment by rickstaa** [comment]
I re-opened this issue so that @qwerty541 and @anuraghazra can give their opinion.

**Comment by qwerty541** [comment]
> I re-opened this issue so that @qwerty541 and @anuraghazra can give their opinion.

@rickstaa Maybe this behavior is good one to be improved but i think i will be very hard to do working with pure SVG. We can return to this after https://github.com/anuraghazra/github-readme-stats/issues/1633 will be implemented.

**Comment by rickstaa** [comment]
> > I re-opened this issue so that @qwerty541 and @anuraghazra can give their opinion.
> 
> @rickstaa Maybe this behavior is good one to be improved but i think i will be very hard to do working with pure SVG. We can return to this after #1633 will be implemented.

I agree let's keep this open for now 👍🏻.

**Comment by airwakz** [comment]
@rickstaa can i do this task and can uh set it for hacktoberfest

**Comment by rickstaa** [comment]
@airwakz of course. Thanks for choosing us for the Hacktoberfest! You can implement the `card_width` and `card_height` arguments for the pin (i.e. repo) card 👍🏻.

**Comment by airwakz** [comment]
`<div align="center">
  <div class="repository-cards">
    <div class="repository-card">
      <a href="https://github.com/rjyo/Air-Test">
        <img src="https://github-readme-stats-git-masterrstaa-rickstaa.vercel.app/api/pin/?username=rjyo&repo=Air-Test"/>
      </a>
    </div>
    <div class="repository-card">
      <a href="https://github.com/meako689/FortyTwoTestTask">
        <img src="https://github-readme-stats-git-masterrstaa-rickstaa.vercel.app/api/pin/?username=meako689&repo=FortyTwoTestTask"/>
      </a>
    </div>
    <div class="repository-card">
      <a href="https://github.com/yourusername/yourrepo">
        <img src="https://github-readme-stats-git-masterrstaa-rickstaa.vercel.app/api/pin/?username=yourusername&repo=yourrepo"/>
      </a>
    </div>
    
  </div>
</div>

<style>
  .repository-cards {
    display: flex;
    justify-content: center;
  }

  .repository-card {
    width: 300px; /* Set the desired width for each card */
    height: 200px; /* Set the desired height for each card */
    margin: 10px; /* Add margin between cards if needed */
  }
</style>`

**Comment by airwakz** [comment]
`<div align="center">
  <div class="repository-cards">
    <div class="repository-card">
      <a href="https://github.com/rjyo/Air-Test">
        <img src="https://github-readme-stats-git-masterrstaa-rickstaa.vercel.app/api/pin/?username=rjyo&repo=Air-Test"/>
      </a>
    </div>
    <div class="repository-card">
      <a href="https://github.com/meako689/FortyTwoTestTask">
        <img src="https://github-readme-stats-git-masterrstaa-rickstaa.vercel.app/api/pin/?username=meako689&repo=FortyTwoTestTask"/>
      </a>
    </div>
    
  </div>
</div>

<style>
  .repository-cards {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
  }

  .repository-card {
    width: 300px; /* Set the desired width for each card */
    height: 200px; /* Set the desired height for each card */
    margin: 10px; /* Add margin between cards if needed */
  }
</style>
`

**Comment by airwakz** [comment]
`<div align="center">
  <div class="repository-cards">
    <div class="repository-card">
      <a href="https://github.com/rjyo/Air-Test">
        <img src="https://github-readme-stats-git-masterrstaa-rickstaa.vercel.app/api/pin/?username=rjyo&repo=Air-Test"/>
      </a>
    </div>
    <div class="repository-card">
      <a href="https://github.com/meako689/FortyTwoTestTask">
        <img src="https://github-readme-stats-git-masterrstaa-rickstaa.vercel.app/api/pin/?username=meako689&repo=FortyTwoTestTask"/>
      </a>
    </div>
    <div class="repository-card">
      <a href="https://github.com/yourusername/yourrepo">
        <img src="https://github-readme-stats-git-masterrstaa-rickstaa.vercel.app/api/pin/?username=yourusername&repo=yourrepo"/>
      </a>
    </div>
    
  </div>
</div>

<style>
  .repository-cards {
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(300px, 1fr)); /* Set card width and responsiveness */
    grid-gap: 10px; /* Adjust the gap between cards */
    justify-content: center;
  }

  .repository-card {
    width: 100%; /* Cards take up the full available width in the grid */
    height: 200px; /* Set the desired height for each card */
  }
</style>
`

**Comment by airwakz** [comment]
`<div align="center"> 
<a href="https://github.com/anuraghazra/github-readme-stats">
<img height=120 src="https://github-readme-stats-git-masterrstaa-rickstaa.vercel.app/api/pin/?username=rjyo&repo=Air-Test"/>
</a>
<a href="https://github.com/anuraghazra/github-readme-stats">
<img height=120 src="https://github-readme-stats-git-masterrstaa-rickstaa.vercel.app/api/pin/?username=meako689&repo=FortyTwoTestTask"/>
</a>
</div>`

**Comment by airwakz** [comment]
ig thanks for the detailed response. I successfully replicated the issue you mentioned. It seems to stem from the use of the HTML height property, causing the SVG image to scale and the fonts to enlarge.

I'm thinking we could introduce a card_height argument, ensuring that it maintains the minimum height required for displaying the text 🤔. Any thoughts on this, @anuraghazra or @qwerty541? Open to better ideas!

Without height property
 
`<div align="center"> 
<a href="https://github.com/anuraghazra/github-readme-stats">
<img src="https://github-readme-stats-git-masterrstaa-rickstaa.vercel.app/api/pin/?username=rjyo&repo=Air-Test"/>
</a>
<a href="https://github.com/anuraghazra/github-readme-stats">
<img src="https://github-readme-stats-git-masterrstaa-rickstaa.vercel.app/api/pin/?username=meako689&repo=FortyTwoTestTask"/>
</a>
</div>
` 
With Height properly
`<div align="center">
  <div class="repository-card">
    <a href="https://github.com/rjyo/Air-Test">
      <img src="https://github-readme-stats-git-masterrstaa-rickstaa.vercel.app/api/pin/?username=rjyo&repo=Air-Test"/>
    </a>
  </div>
  <div class="repository-card">
    <a href="https://github.com/meako689/FortyTwoTestTask">
      <img src="https://github-readme-stats-git-masterrstaa-rickstaa.vercel.app/api/pin/?username=meako689&repo=FortyTwoTestTask"/>
    </a>
  </div>
</div>

<style>
  .repository-card {
    width: 300px; /* Set the desired width for each card */
    height: 200px; /* Set the desired height for each card */
    margin: 10px; /* Add margin between cards if needed */
  }
</style>`

**Comment by airwakz** [comment]
@rickstaa do uh want me to make any changes to

**Comment by rickstaa** [comment]
@airwakz, your conclusion is correct (see https://github.com/anuraghazra/github-readme-stats/issues/3159#issuecomment-1761169385). The same problem occurs with the other cards in https://github.com/anuraghazra/github-readme-stats/issues/3159. Adding the `card_width` and `card_height` parameters to the pin card will solve the problem. https://github.com/anuraghazra/github-readme-stats/issues/2900#issuecomment-1617583853

I already assigned @anmolchhabra21 to solve this problem for the cards. I [just asked](https://github.com/anuraghazra/github-readme-stats/issues/3159#issuecomment-1762778407) if he is still working on that issue. In the meantime, you can add the `card_width` parameter to the pin card and create a separate pull request for that (see https://github.com/anuraghazra/github-readme-stats/issues/2900#issuecomment-1617583853).

**Comment by airwakz** [comment]
@rickstaa can uh tell me in which file i need to make changes as i am unable to make changes in the file and a pr need a change of html code or  a file

**Comment by rickstaa** [comment]
Yea, no problem. To implement the `card_width` argument, you need to add the `card_width` code found in the stats card (i.e. https://github.com/anuraghazra/github-readme-stats/blob/master/src/cards/stats-card.js) to the pin card (i.e. https://github.com/anuraghazra/github-readme-stats/blob/master/src/cards/repo-card.js). It would be mainly JavaScript code. You can check out the https://github.com/anuraghazra/github-readme-stats/blob/master/CONTRIBUTING.md#local-development guide on how you can run and debug your changes on your local machine.

**Comment by rickstaa** [comment]
@airwakz I just noticed that the contributing guide is outdated and created a pull request -> https://github.com/anuraghazra/github-readme-stats/pull/3358/files. You can use the steps in this pull request.

**Comment by airwakz** [comment]
`/ Import the card_width logic from the "stats card"
import { CARD_MIN_WIDTH as REPO_CARD_MIN_WIDTH, CARD_DEFAULT_WIDTH as REPO_CARD_DEFAULT_WIDTH } from "./stats-card.js";

// ... (Existing code)

// Define your own minCardWidth and defaultCardWidth for the "repo card"
const REPO_CARD_ICON_SIZE = 16; // Adjust this value as needed
const REPO_CARD_MIN_WIDTH = /* Calculate the minimum width for the repo card */;
const REPO_CARD_DEFAULT_WIDTH = /* Calculate the default width for the repo card */;

/**
 * Check the card_width option and set the card width accordingly.
 */
const cardWidth = options.card_width
  ? isNaN(options.card_width)
    ? REPO_CARD_DEFAULT_WIDTH
    : Math.max(REPO_CARD_MIN_WIDTH, options.card_width) // Ensure it doesn't go below the minimum width
  : REPO_CARD_DEFAULT_WIDTH;

// ... (Existing code)

/**
 * Create the Card instance with the calculated cardWidth.
 */
const card = new Card({
  defaultTitle: header.length > 35 ? `${header.slice(0, 35)}...` : header,
  titlePrefixIcon: icons.contribs,
  width: cardWidth, // Use the calculated card width
  height,
  border_radius,
  colors,
});

// ... (Remaining code)`
this is the code that i have written can i make changes with this code  @rickstaa

**Comment by razonyang** [comment]
@airwakz Thank you spending time on this, I'm curious how it look likes in the case of multiple rows. i.e. the GitHub example below, the `1` and `3` cards take 1 line for descriptions, and the rest of cards take 2 lines. Would it look like the example (equal height and width for all cards) after fixing this issue?

![image](https://github.com/anuraghazra/github-readme-stats/assets/17720932/5beccc3c-adb7-4da4-94cc-661430217fb8)

**Comment by airwakz** [comment]
@razonyang yah  i kept original 6 in mind before solving this issue as we can select max 6 repos

**Comment by rickstaa** [comment]
> `/ Import the card_width logic from the "stats card" import { CARD_MIN_WIDTH as REPO_CARD_MIN_WIDTH, CARD_DEFAULT_WIDTH as REPO_CARD_DEFAULT_WIDTH } from "./stats-card.js";
> 
> // ... (Existing code)
> 
> // Define your own minCardWidth and defaultCardWidth for the "repo card" const REPO_CARD_ICON_SIZE = 16; // Adjust this value as needed const REPO_CARD_MIN_WIDTH = /* Calculate the minimum width for the repo card _/; const REPO_CARD_DEFAULT_WIDTH = /_ Calculate the default width for the repo card */;
> 
> /**
> 
> * Check the card_width option and set the card width accordingly.
>   */
>   const cardWidth = options.card_width
>   ? isNaN(options.card_width)
>   ? REPO_CARD_DEFAULT_WIDTH
>   : Math.max(REPO_CARD_MIN_WIDTH, options.card_width) // Ensure it doesn't go below the minimum width
>   : REPO_CARD_DEFAULT_WIDTH;
> 
> // ... (Existing code)
> 
> /**
> 
> * Create the Card instance with the calculated cardWidth.
>   */
>   const card = new Card({
>   defaultTitle: header.length > 35 ? `${header.slice(0, 35)}...` : header,
>   titlePrefixIcon: icons.contribs,
>   width: cardWidth, // Use the calculated card width
>   height,
>   border_radius,
>   colors,
>   });
> 
> // ... (Remaining code)` this is the code that i have written can i make changes with this code @rickstaa

@airwakz Can you maybe create a pull request? This will make it a lot easier for me to review your code.f

**Comment by rickstaa** [comment]
> @airwakz Thank you spending time on this, I'm curious how it look likes in the case of multiple rows. i.e. the GitHub example below, the `1` and `3` cards take 1 line for descriptions, and the rest of cards take 2 lines. Would it look like the example (equal height and width for all cards) after fixing this issue?
> 
> ![image](https://user-images.githubusercontent.com/17720932/275214119-5beccc3c-adb7-4da4-94cc-661430217fb8.png)

I think the width and height should be handled by the user through the `card_width` and `card_height` arguments. This allows people to create the kind of grid they want.

**Comment by airwakz** [comment]
Now, users can specify the card_width and card_height when using the renderRepoCard function to control the dimensions of the repository card. If not provided, it falls back to default value and can uh tell me its default value 
@rickstaa @razonyang

**Comment by airwakz** [comment]
@rickstaa  opened a pr  just check it out

**Comment by airwakz** [comment]
![image](https://github.com/anuraghazra/github-readme-stats/assets/114548361/dcc9fa75-68fb-45e9-983e-e431d52e374d)

**Comment by airwakz** [comment]
@qwerty541 @rickstaa https://github.com/anuraghazra/github-readme-stats/pull/3359 i have raised a pull request kindly merge it

**Comment by qwerty541** [comment]
Pull request #3453 was merged, now equal height for different repositories cards can be achieved by the following way:

## Before:

```html
<div align="center"> 
<a href="https://github.com/anuraghazra/github-readme-stats">
<img src="https://github-readme-stats.vercel.app/api/pin/?username=rjyo&repo=Air-Test"/>
</a>
<a href="https://github.com/anuraghazra/github-readme-stats">
<img src="https://github-readme-stats.vercel.app/api/pin/?username=meako689&repo=FortyTwoTestTask"/>
</a>
</div>
```

<div align="center"> 
<a href="https://github.com/anuraghazra/github-readme-stats">
<img src="https://github-readme-stats.vercel.app/api/pin/?username=rjyo&repo=Air-Test"/>
</a>
<a href="https://github.com/anuraghazra/github-readme-stats">
<img src="https://github-readme-stats.vercel.app/api/pin/?username=meako689&repo=FortyTwoTestTask"/>
</a>
</div>

## After (`&description_lines_count=3`):

```html
<div align="center"> 
<a href="https://github.com/anuraghazra/github-readme-stats">
<img src="https://github-readme-stats.vercel.app/api/pin/?username=rjyo&repo=Air-Test&description_lines_count=3"/>
</a>
<a href="https://github.com/anuraghazra/github-readme-stats">
<img src="https://github-readme-stats.vercel.app/api/pin/?username=meako689&repo=FortyTwoTestTask&description_lines_count=3"/>
</a>
</div>
```

<div align="center"> 
<a href="https://github.com/anuraghazra/github-readme-stats">
<img src="https://github-readme-stats.vercel.app/api/pin/?username=rjyo&repo=Air-Test&description_lines_count=3"/>
</a>
<a href="https://github.com/anuraghazra/github-readme-stats">
<img src="https://github-readme-stats.vercel.app/api/pin/?username=meako689&repo=FortyTwoTestTask&description_lines_count=3"/>
</a>
</div>

## After (`&description_lines_count=1`):

```html
<div align="center"> 
<a href="https://github.com/anuraghazra/github-readme-stats">
<img src="https://github-readme-stats.vercel.app/api/pin/?username=rjyo&repo=Air-Test&description_lines_count=1"/>
</a>
<a href="https://github.com/anuraghazra/github-readme-stats">
<img src="https://github-readme-stats.vercel.app/api/pin/?username=meako689&repo=FortyTwoTestTask&description_lines_count=1"/>
</a>
</div>
```

<div align="center"> 
<a href="https://github.com/anuraghazra/github-readme-stats">
<img src="https://github-readme-stats.vercel.app/api/pin/?username=rjyo&repo=Air-Test&description_lines_count=1"/>
</a>
<a href="https://github.com/anuraghazra/github-readme-stats">
<img src="https://github-readme-stats.vercel.app/api/pin/?username=meako689&repo=FortyTwoTestTask&description_lines_count=1"/>
</a>
</div>

**Comment by qoomon** [comment]
Would be awesome if `&description_lines_count` will be implemented for gist cards as well
